test/boot/associated_object.rb

```diff
@@ -55,3 +55,33 @@ def great?
 end
 
 Post::Comment.has_object :rating
+
+# Can locate subclasses by grepping for `< Pricing`
+class Pricing < ActiveRecord::AssociatedObject::Polymorphic
+  # Here, `record` will return the post for Post::Pricing objects and the comment for Post::Comment::Pricings.
+
+  def something_conditional
+    execute_behavior if post?
```

Comment:
I'm trying to figure if it makes sense to add these `post?` and `comment?` shorthand methods to allow conditional behavior in the polymorphic parent class. Or if it's better to just rely on OOP and define a basic implementation in the polymorphic parent class that the subclasses can still enrich.

Reply:
It's always tough to know without using it in production code, but my gut says this would lead to wonky things. I like the idea of keeping it 100% shareable so the associated object can focus only on the things that are universal to all of the classes that will use it.

Reply:
Yeah, that does make sense to me and I think you're right about skipping that support for `post?`/`comment?`.